build_modules/lib/src/module_cache.dart

```diff
@@ -53,9 +60,26 @@ class DecodingCache<T> {
   /// content dependencies will be tracked through [reader].
   Future<T> find(AssetId id, AssetReader reader) async {
     if (!await reader.canRead(id)) return null;
-    var result = _cached.putIfAbsent(
-        id, () => Result.capture(reader.readAsBytes(id).then(_fromBytes)));
-    return Result.release(result);
+    if (_cached.containsKey(id)) {
+      final entry = _cached[id];
+      if (entry.needsCheck) {
+        final digest = await reader.digest(id);
+        if (digest != entry.digest) {
+          entry
+            ..needsCheck = false
+            ..digest = digest
+            ..value = Result.capture(reader.readAsBytes(id).then(_fromBytes));
+        } else {
+          entry.needsCheck = false;
```

Comment:
nit: move this out of the else and don't assign it in the if (it should unconditionally be done)